lib/chatter_web/live/chat_live/index.ex

```diff
@@ -111,4 +137,72 @@ defmodule ChatterWeb.ChatLive.Index do
       "=" -> Enum.count(message.likes) == likes_count
     end
   end
+
+  defp filter_messages_by_option(option) do
+    case option do
+      :with_likes_who_liked -> filter_with_likes_who_like()
+      :without_likes_who_never_liked -> filter_without_likes_who_never_liked()
+      :with_major_likes -> filter_with_major_likes()
+    end
+  end
+
+  defp filter_with_likes_who_like do
+    Enum.filter(MessageAgent.get(), fn message ->
+      if Enum.count(message.likes) > 0 && message.author in MessageAgent.get_all_likes(),
```

Comment:
`N+1` problem here
it gets all likes authors for each message

can be improved

Reply:
right, getting all likes is moved to the top of function